index.js

```diff
@@ -4,23 +4,85 @@ var fs = require('fs');
 var backlightPath = '/sys/class/backlight/rpi_backlight';
 
 function writeValue(fileName, value) {
-    return new Promise(function (resolve, reject) {
-        var fullPath = path.join(backlightPath, fileName);
-        fs.writeFile(fullPath, value, function(err) {
-             if (err !== null) reject(err);
-             else resolve();
-        });
+  return new Promise((resolve, reject) => {
+    var fullPath = path.join(backlightPath, fileName);
+    fs.writeFile(fullPath, value, (err) => {
+      if (err !== null) reject(err);
+      else resolve();
     });
+  });
+}
+
+function readValue(fileName, value) {
+  return new Promise((resolve, reject) => {
+    var fullPath = path.join(backlightPath, fileName);
+    fs.readFile(fullPath, 'utf8', (err, data) => {
+      if (err !== null){
+        reject(err); 
+      } else {
+        resolve(data);
+      }
+    });
+  });
 }
 
 if (!fs.existsSync(backlightPath)) {
-    throw new Error('Backlight control not supported (' + backlightPath + ' does not exist)');
+  throw new Error('Backlight control not supported (' + backlightPath + ' does not exist)');
 }
 
-exports.powerOn = function () {
-    return writeValue('bl_power', '0');
+/** Power managment */
+exports.powerOn = () => {
+  return writeValue('bl_power', '0');
+};
+
+exports.powerOff = () => {
+  return writeValue('bl_power', '1');
+};
+
+exports.getActualPowerValue = () => {
+  return readValue('bl_power');
+};
+
+/** Brightness managment */
+exports.setMaxBrightness = () => {
+  return writeValue('brightness', '255');
 };
 
-exports.powerOff = function () {
-    return writeValue('bl_power', '1');
+exports.setBrightnessValue = (value) => {
+  if (value > 255 || value < 0) {
+    return "ERR: Incorrect value";
+  }
+  return writeValue('brightness', value);
 };
+
+exports.goToBrightnessValue = (toValue, speed) => {
+  if (toValue > 255 || toValue < 0) {
+    return "ERR: Incorrect value";
+  }
+  let from = this.getActualBrightness();
+  from.then((data) => {
+    if (toValue < data) {
+      let goDown = setInterval(function(){ 
+        if (data > toValue) {
+          writeValue('brightness', data--); 
+        } else {
+          clearInterval(goDown);
+        }                   
+      }, speed);
+    } else if (toValue > data) {
+      let goUp = setInterval(function(){ 
+        if (data < toValue) {
+          writeValue('brightness', data++); 
+        } else {
+          clearInterval(goUp);
+        }                   
+      }, speed);
+    }
+  }); 
+  
+  return "OK";
+};
```

Comment:
I think you said this isn't really ready yet? I'd leave it out of this PR, and do it separately later.

Reply:
Yes indeed, but I did not really know what had to return the function ... I think just nothing, a boolean or the final brightness value to confirm that the operation went well ? 

I will pay more attention to the future to have a cleaner PR.

Reply:
For now I'd leave it out completely.

We can discuss the return type in more depth in a later PR, but if you don't have a meaningful return value, I'd return nothing. Here, if you really might have success or failure I'd return actual `true`/`false` booleans, and if you have a process that progresses and eventually finishes, I'd use a promise: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises (maybe even a promise that _eventually_ becomes a boolean). Let's leave this for a later PR though and keep things simple for now.